content/algorithms/dag/index.md

```diff
@@ -0,0 +1,216 @@
+    with_labels=True,
+    node_size=1000,
+    node_color="#ffff8f",
+    width=0.8,
+    font_size=14,
+)
+```
+
+### Definition
+
+In mathematics, and more specifically in graph theory,
+a directed graph (or DiGraph) is a graph that is made up of a set of vertices
+connected by directed edges often called arcs.
+
+## Directed Acyclic Graph
+
+### Example
+
+```{code-cell} ipython3
+# load Professor Bumstead clothing graph
+clothing_graph = nx.read_graphml(f"data/clothing_graph.graphml")
+```
+
+```{code-cell} ipython3
+# set figure size
+plt.figure(figsize=(12, 12), dpi=150)
+
+# draw `clothing_graph`
+nx.draw_planar(clothing_graph,
+    arrowsize=12,
+    with_labels=True,
+    node_size=8000,
+    node_color="#ffff8f",
+    linewidths=2.0,
+    width=1.5,
+    font_size=14,
+)
+```
+
+Here is an example that arises when Professor Bumstead gets dressed in the morning.
+The professor must do certain garments before others (e.g., socks before shoes).
+Other items may be put on in any order (e.g., socks and pants).
+
+A directed edge $(u, v)$ in the example indicates that garment $u$
+must be donned before garment $v$.
+
+For example, the `clothing_graph` is a DAG.
+
+```{code-cell} ipython3
+# check if the `clothing_graph` is DAG
+print(nx.is_directed_acyclic_graph(clothing_graph))
+```
+
+But, the `triangle_graph` is not a DAG.
+
+```{code-cell} ipython3
+# check if the `triangle_graph` is DAG
+print(nx.is_directed_acyclic_graph(triangle_graph))
+```
+
+`triangle_graph` has a cycle:
+
+```{code-cell} ipython3
+# find cycle in the `triangle_graph`
+print(nx.find_cycle(triangle_graph))
+```
+
+### Applications
+
+Directed acyclic graphs representations of partial orderings have many applications in scheduling
+for systems of tasks with ordering constraints.
+An important class of problems of this type concern collections of objects that need to be updated,
+such as the cells of a spreadsheet after one of the cells has been changed,
+or the object files of a piece of computer software after its source code has been changed.
+In this context, a dependency graph is a graph that has a vertex for each object to be updated,
+and an edge connecting two objects whenever one of them needs to be updated earlier than the other.
+A cycle in this graph is called a circular dependency, and is generally not allowed,
+because there would be no way to consistently schedule the tasks involved in the cycle.
+Dependency graphs without circular dependencies form DAGs.
+
+A directed acyclic graph may be used to represent a network of processing elements.
+In this representation, data enters a processing element through its incoming edges
+and leaves the element through its outgoing edges.
+For instance, in electronic circuit design, static combinational logic blocks
+can be represented as an acyclic system of logic gates that computes a function of an input,
+where the input and output of the function are represented as individual bits.
```

Comment:
This is a really nice example that highlights the *directed* part of the graph, but is there something more specific in circuit design to highlight the *acyclic* part? In my experience, feedback plays an important part in circuit design, so it might not be the best fit to highlight directed *acyclic* graphs --- unless you had a specific design in mind where cycles are expressly prohibited.